lib/faraday/connection.rb

```diff
@@ -464,5 +442,14 @@ def set_authorization_header(header_type, *args)
         header(*args)
       headers[Faraday::Request::Authorization::KEY] = header
     end
+
+    def find_default_proxy
+      warn 'no_proxy is unsupported' if ENV['no_proxy'] || ENV['NO_PROXY']
```

Comment:
So no_proxy is unsupported when the url is not given upfront in the initializer, but later on in the request?